test/test_time.cpp

```diff
@@ -26,6 +28,20 @@ inline void res( CLOCKID clk_id )
             << std::endl;
 }
 
+static void* test_thread_time(void *arg)
```

Comment:
Why make the argument a `void *` when what you pass is just an int and you anyway cast back to int before using. Just pass as a int by-value and avoid all the ugly `void*` and casting.

Reply:
That's the function signature required by `pthread_create`.

Reply:
I not sure that cast `void*` to `int` makes things clear. It still ugly cast.

And as Joshua pointed it is the signature which enforces pointers.